src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,423 @@
+	cvrf_element_add_child("DocumentTitle", cvrf_model_get_doc_title(session->model), root_node);
+	cvrf_element_add_child("DocumentType", cvrf_model_get_doc_type(session->model), root_node);
+	//cvrf_export_element(os_name, "OS Name", writer);
+
+	struct cvrf_vulnerability_iterator *it = cvrf_model_get_vulnerabilities(session->model);
+	bool vulnerable = false;
+	while (cvrf_vulnerability_iterator_has_more(it)) {
+		struct cvrf_vulnerability *vuln = cvrf_vulnerability_iterator_next(it);
+		xmlNode *vuln_node = xmlNewTextChild(root_node, NULL, BAD_CAST "Vulnerability", NULL);
+
+		cvrf_element_add_child("Title", cvrf_vulnerability_get_title(vuln), vuln_node);
+		cvrf_element_add_child("ID", cvrf_vulnerability_get_system_id(vuln), vuln_node);
+		cvrf_element_add_child("DiscoveryDate", cvrf_vulnerability_get_discovery_date(vuln), vuln_node);
+		cvrf_element_add_child("ReleaseDate", cvrf_vulnerability_get_release_date(vuln), vuln_node);
+		cvrf_element_add_child("CVE", cvrf_vulnerability_get_cve_id(vuln), vuln_node);
+
+		xmlNode *statuses_node = xmlNewTextChild(vuln_node, NULL, BAD_CAST "ProductStatuses", NULL);
+		struct oscap_string_iterator *product_ids = cvrf_session_get_product_ids(session);
+		while (oscap_string_iterator_has_more(product_ids)) {
+			const char *product_id = oscap_string_iterator_next(product_ids);
+			/*
+			if (cvrf_product_vulnerability_fixed(vuln, product_id)) {
+
+			}
+			else {
+
+				vulnerable = true;
+			}*/
+		}
+
+		oscap_string_iterator_free(product_ids);
+		vulnerable = false;
+	}
+	cvrf_vulnerability_iterator_free(it);
+
+	struct oscap_source *source = oscap_source_new_from_xmlDoc(doc, export_file);
+	int ret = oscap_source_save_as(source, NULL);
+	oscap_source_free(source);
+	cvrf_session_free(session);
+	return ret;
+}
+
+
+static const char *get_rpm_name_from_cvrf_product_id(struct cvrf_session *session, const char *product_id) {
+	const char *rpm_name = NULL;
+	struct cvrf_product_tree *tree = cvrf_model_get_product_tree(session->model);
+
+	struct oscap_iterator *branches = cvrf_product_tree_get_branches(tree);
+	while (oscap_iterator_has_more(branches)) {
+		struct cvrf_branch *branch = oscap_iterator_next(branches);
+		if (cvrf_branch_get_branch_type(branch) == CVRF_BRANCH_PRODUCT_VERSION) {
+			struct cvrf_product_name *full_name = cvrf_branch_get_cvrf_product_name(branch);
+
+			if (oscap_str_endswith(product_id, cvrf_product_name_get_product_id(full_name))) {
+				rpm_name = cvrf_product_name_get_cpe(full_name);
+				break;
+			}
+		}
+	}
+	oscap_iterator_free(branches);
+	return rpm_name;
+}
+
+static struct cvrf_rpm_attributes *parse_rpm_attributes_from_cvrf_product_id(struct cvrf_session *session,
+		const char *product_id) {
+
+	struct cvrf_rpm_attributes *attributes = cvrf_rpm_attributes_new();
+	attributes->full_package_name = strdup(get_rpm_name_from_cvrf_product_id(session, product_id));
+	char *package = strdup(strchr(product_id, ':')+1);
+	unsigned int index = 0;
```

Comment:
also, can't you just use strchr instead of that loop? the compiler will most likely figure out what you are doing and do it faster but it would make it easier for the person reading the code.